src/sentry/api/event_search.py

```diff
@@ -552,7 +555,14 @@ def _handle_numeric_filter(self, search_key, operator, search_value):
                 raise InvalidSearchQuery(str(exc))
             return SearchFilter(search_key, operator, search_value)
 
-        return self._handle_text_filter(search_key, operator, SearchValue("".join(search_value)))
+        search_value = SearchValue("".join(search_value))
+        if operator not in ("=", "!=") and search_key.name not in self.config.text_operator_keys:
+            # Operators are not supported in text_filter.
+            # Push it back into the value before handing the negation.
+            search_value = search_value._replace(raw_value=f"{operator}{search_value.raw_value}")
+            operator = "!=" if negated else "="
```

Comment:
Hm, I just realized most of this logic happens in `self._handle_text_filter` but it only sets `operator = "="`. I'd like to remove this duplication.